Develop2

GNailya opened the pull request:
Написала тесты на регистрацию пользователя, авторизацию, обновление данных, создание заказа, получение списка заказов

nelyagabdulkhakova committed:
add allure report

nelyagabdulkhakova committed:
Написала тесты на регистрацию пользователя, авторизацию, обновление данных, создание заказа, получение списка заказов

pomegranate282922 commented on src/main/java/site/nomoreparties/stellarburgers/RestAssuredClient.java:
```diff
@@ -0,0 +1,16 @@
+    public static RequestSpecification getSpec() {
+        return new RequestSpecBuilder()
+                .log(LogDetail.ALL)
+                .setContentType(ContentType.JSON)
```
Сюда вставь ещё base uri и тогда массу повторов base url можно убрать из вызовов

pomegranate282922 commented on src/main/java/site/nomoreparties/stellarburgers/model/User.java:
```diff
@@ -0,0 +1,51 @@
+import org.apache.commons.lang3.RandomStringUtils;
+
+public class User {
+    public String email;
```
Нарушение принципа инкапсуляции. Поля - приватные. Исправь везде.

pomegranate282922 commented on src/test/java/site/nomoreparties/stellarburgers/CreateWithEmptyFieldTest.java:
```diff
@@ -0,0 +1,59 @@
+        assertThat(statusCode, equalTo(403));
+        assertThat(message, equalTo("Email, password and name are required fields"));
+
+    }
```
Если поймается баг и юзер -таки создастся, его тоже нужно удалить

pomegranate282922 commented on src/test/java/site/nomoreparties/stellarburgers/UpdateUserTest.java:
```diff
@@ -0,0 +1,106 @@
+public class UpdateUserTest {
+    private UserClient userClient;
+    private User user;
+    String accessToken;
```
А почему это поле без модификатора доступа?

pomegranate282922 reviewed (commented):
Хорошо! Небольшие поправочки прошу внести

nelyagabdulkhakova committed:
- Добавила  base uri в спецификации,
- исправила/добавила модификаторы,
- добавила удаление пользователя в тестах по созданию пользователя с пустыми полями,
- части урлов вынесла в константы

pomegranate282922 commented on src/main/java/site/nomoreparties/stellarburgers/RestAssuredClient.java:
```diff
@@ -0,0 +1,16 @@
+        return new RequestSpecBuilder()
+                .log(LogDetail.ALL)
+                .setContentType(ContentType.JSON)
+                .setBaseUri("https://stellarburgers.nomoreparties.site/api/")
```
Лучше ещё урл вынести как константу- нагляднее будет.

pomegranate282922 reviewed (approved):
Отлично!
